### PR TITLE
 feat(code-editor): create code-editor component

### PR DESCRIPTION
**Pull request by marinasmah**
Create new component "Code Editor", according to [Spec](https://jira-eng-sjc12.cisco.com/jira/browse/CAX-204)
## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/code-editor/scss/module.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+@import "@/wc_scss/tools/mixins/core";
+
+@import "@/wc_scss/settings/rem";
+@import "@/wc_scss/colors/settings";
+@import "@/wc_scss/settings/core";
+
+@import "@/wc_scss/typography/settings";
+@import "@/wc_scss/tools/mixins/flex";
+
+//@import "./vars/-settings";
```
Comment (futurethang) [inline web-components/src/components/code-editor/scss/module.scss:11]: nit: remove

**Review by futurethang — CHANGES_REQUESTED** [review]

**Comment by futurethang** [comment]
Kind of a side note, but where can I find the design doc for this? Is it a Figma collection?

File: web-components/src/components/code-editor/scss/module.scss (hunk 1 above)
Comment (marinasmah) [reply web-components/src/components/code-editor/scss/module.scss]: fixed

**Comment by marinasmah** [comment]
> Kind of a side note, but where can I find the design doc for this? Is it a Figma collection?

@futurethang plz look at link in PR Description

File: web-components/src/components/code-editor/CodeEditor.stories.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  parameters: {
+    a11y: {
+      element: "md-code-editor"
+    }
+  }
+};
+
+export const CodeEditor = () => {
+  return html`
+    <md-code-editor></md-code-editor>
```
Comment (elenagiere) [inline web-components/src/components/code-editor/CodeEditor.stories.ts:19]: I would add the properties as knobs to this storybook component:
acceptLanguage, getLocalization, copyLocalization, copiedLocalization

Then run `yarn storybook` locally to test codeEditor in storybook

File: web-components/src/components/code-editor/scss/settings.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+$md-code-editor-height: rem-calc(300);
+$md-code-editor-indent: rem-calc(10);
+$md-code-editor-color: $md-white-100;
+$md-code-editor-bg: $md-blue-95;
+$md-code-editor-header-bg: $md-blue-90;
```
Comment (elenagiere) [inline web-components/src/components/code-editor/scss/settings.scss:5]: I noticed all the color definitions are hard coded and that there aren't light/dark colors defined specifically for this component. Is that intentional?

**Review by elenagiere — COMMENTED** [review]
Left 2 comments.

Also, when you add a new component to the sandbox, try to do so in alphabetical order.

File: web-components/src/components/code-editor/CodeEditor.stories.ts (hunk 2 above)
Comment (marinasmah) [reply web-components/src/components/code-editor/CodeEditor.stories.ts]: I added only acceptLanguage prop. the others are not needed for storybook

File: web-components/src/components/code-editor/scss/settings.scss (hunk 3 above)
Comment (marinasmah) [reply web-components/src/components/code-editor/scss/settings.scss:5]: We have just one color theme for code editor

**Review by futurethang — DISMISSED** [review]

File: web-components/src/components/code-editor/CodeEditor.ts [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  @internalProperty() private fileName = "";
+  @internalProperty() private copied = false;
+
+  @queryAssignedNodes("code-block") slotNodes!: Node[];
+
+  static get styles() {
+    return [reset, styles];
+  }
+
+  triggerFileLoad(event: MouseEvent) {
```
Comment (yananym) [inline web-components/src/components/code-editor/CodeEditor.ts:39]: Get is not a button - GET is a type of API request/response for API reference, It simply should be a badge with variation for GET/POST

**Comment by yananym** [comment]
Please add more use cases to the sandbox for reference with different language syntax

**Review by yananym — APPROVED** [review]

**Pull request merged**